src/typeguard/_transformer.py

```diff
@@ -375,6 +375,16 @@ def visit_BinOp(self, node: BinOp) -> Any:
         self.generic_visit(node)
 
         if isinstance(node.op, BitOr):
+            # If either branch of the BinOp has been transformed to `None`
+            # then the `ast.generic_visit` will eliminate that branch completely.
+            # If this happens, treat the BinOp as just the other branch.
+            if not hasattr(node, "left") and not hasattr(node, "right"):
+                return None
+            if not hasattr(node, "left"):
+                return node.right
+            if not hasattr(node, "right"):
```

Comment:
This is just for aesthetics, but I prefer separate control blocks to be separated by a blank line.
```suggestion
            elif not hasattr(node, "left"):
                return node.right
            elif not hasattr(node, "right"):
```